pkg/integrations/gcp_exporter/gcp_exporter.go

```diff
@@ -0,0 +1,214 @@
+				IngestDelay:           c.IngestDelay,
+				DropDelegatedProjects: c.DropDelegatedProjects,
+
+				// If FillMissingLabels ensures all metrics with the same name have the same label set. It's not often
+				// that GCP metrics have different labels but if it happens the prom registry will panic.
+				FillMissingLabels: true,
+
+				// If AggregateDeltas is disabled the data produced is not useful at all. See https://github.com/prometheus-community/stackdriver_exporter#what-to-know-about-aggregating-delta-metrics
+				// for more info
+				AggregateDeltas: true,
+			},
+			l,
+			collectors.NewInMemoryDeltaCounterStore(l, 30*time.Minute),
+			collectors.NewInMemoryDeltaDistributionStore(l, 30*time.Minute),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create monitoring collector: %w", err)
+		}
+		gcpCollectors = append(gcpCollectors, monitoringCollector)
+	}
+
+	return integrations.NewCollectorIntegration(
+		c.Name(), integrations.WithCollectors(gcpCollectors...),
+	), nil
+}
+
+func (c *Config) Validate() error {
+	var configErrors []string
+
+	if c.ProjectIDs == nil || len(c.ProjectIDs) == 0 {
+		configErrors = append(configErrors, "no project_ids defined and no default was found in the gcloud config")
+	}
+
+	if c.MetricPrefixes == nil || len(c.MetricPrefixes) == 0 {
+		configErrors = append(configErrors, "at least 1 metrics_prefixes is required")
+	}
+
+	if len(c.ExtraFilters) > 0 {
+		filterPrefixToFilter := map[string][]string{}
+		for _, filter := range c.ExtraFilters {
+			splitFilter := strings.Split(filter, ":")
+			if len(splitFilter) <= 1 {
+				configErrors = append(configErrors, fmt.Sprintf("%s is an invalid filter a filter must be of the form <metric_type>:<filter_expression>", filter))
+				continue
+			}
+			filterPrefix := splitFilter[0]
+			if _, exists := filterPrefixToFilter[filterPrefix]; !exists {
+				filterPrefixToFilter[filterPrefix] = []string{}
+			}
+			filterPrefixToFilter[filterPrefix] = append(filterPrefixToFilter[filterPrefix], filter)
+		}
+
+		for filterPrefix, filters := range filterPrefixToFilter {
+			validFilterPrefix := false
+			for _, metricPrefix := range c.MetricPrefixes {
+				if strings.HasPrefix(metricPrefix, filterPrefix) {
+					validFilterPrefix = true
+					break
+				}
+			}
+			if !validFilterPrefix {
+				configErrors = append(configErrors, fmt.Sprintf("No metric_prefixes started with %s which means the extra_filters %s will not have any effect", filterPrefix, strings.Join(filters, ",")))
+			}
+		}
+	}
+
+	if len(configErrors) != 0 {
+		return errors.New(strings.Join(configErrors, ","))
+	}
+
+	return nil
+}
+
+func createMonitoringService(ctx context.Context, httpTimeout time.Duration) (*monitoring.Service, error) {
+	googleClient, err := google.DefaultClient(ctx, monitoring.MonitoringReadScope)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Google client: %v", err)
+	}
+
+	googleClient.Timeout = httpTimeout
+	googleClient.Transport = rehttp.NewTransport(
```

Comment:
I didn't look deeper, but you reckon we could use [this](https://github.com/hashicorp/go-retryablehttp) instead? It's already in the dependencies and seems to be more popular than `rehttp`.

Reply:
I gave it a shot like,

```go
	googleClient, err := google.DefaultClient(ctx, monitoring.MonitoringReadScope)
	if err != nil {
		return nil, fmt.Errorf("error creating Google client: %v", err)
	}

	retryClient := retryablehttp.NewClient()
	retryClient.RetryMax = 4
	retryClient.RetryWaitMin = time.Second
	retryClient.RetryWaitMax = 5 * time.Second
	retryClient.Logger = nil

	googleClient.Transport = &retryablehttp.RoundTripper{Client: retryClient}

	monitoringService, err := monitoring.NewService(ctx,
		option.WithHTTPClient(googleClient),
	)
```

Unfortunately, the end result seemed to disable the auth from the `google.DefaultClient` 😢